packages/smooth_app/lib/helpers/text_input_formatters_helper.dart

```diff
@@ -10,14 +10,43 @@ import 'package:smooth_app/helpers/strings_helper.dart';
 /// on the user's language.
 ///
 /// It works if there are no group separator in the format, as it would be
-/// problematic with the `123,456.9` and both comma and dot.
+/// problematic with the `1,234,567.89` and both comma and dot.
 ///
 /// Also, if a separator is already displayed, it will be moved to the new
 /// position
 class DecimalSeparatorRewriter extends TextInputFormatter {
   DecimalSeparatorRewriter(NumberFormat format)
       : _decimalSeparator = format.symbols.DECIMAL_SEP,
-        _separatorToReplace = _findSeparatorToReplace(format);
+        _separatorToReplace = _findSeparatorToReplace(format) {
+    // Here we check that there are no group separators.
```

Comment:
I'm not familiar with Dart, so excuse me if this is a dumb question. Are those tests run inside the constructor each time an object is created? That seems a bit weird, why not put them only in unit tests instead?

Reply:
That's correct, each time an object is created this will be run. Here the idea is to prevent the app from working with "wrong number formats", at run-time, when the user changed the device language and the app language.
Could we run unit tests instead? That would be another test, interesting too: pre-checking at build time that the number format is correct.

Reply:
I tried to go up a bit, do I understand correctly that in Smoothie, the format is always this one?

```
  static NumberFormat getNumberFormat({required final bool decimal}) =>
      NumberFormat(
        decimal ? '####0.#####' : '######0',
        ProductQuery.getLocaleString(),
      );
```

So there is no way that we would get a string separator by calling format.format(number), right?

Reply:
There's no way we get a group separator in format.format, unless some day some UI "improvement" actually adds a group separator. That was the point of this PR: an immediate app crash when developing.

We have a numeric keyboard and an automatic rewrite: there will be only 1 decimal separator (even if the user types another occurrence of the same separator), and no group separator.

Reply:
Sounds good, thanks.